src/commands/serverActions.ts

```diff
@@ -184,6 +184,7 @@ export async function serverActions(): Promise<void> {
             const path = vscode.Uri.parse(urlString).path;
```

Comment:
Instead of manipulating the URL as a string and then parsing it into a `vscode.Uri`, why not parse it from the start and then manipulate it as an object? I think code like this is much better:

```typescript
          let url = vscode.Uri.parse(action.detail);
          if (action.rawLink?.startsWith("${serverUrl}")) {
            const token = await getCSPToken(api, url.path);
            if (token.length > 0) {
              url = url.with({
                query: url.query.length ? `${url.query}&CSPCHD=${token}` : `CSPCHD=${token}`
              });
            }
          }
          vscode.env.openExternal(url);
```